consensus/pow.py

```
@@ -0,0 +1,68 @@
+    block.nonce = 0
+
+    if logger:
+        logger.info(f"Mining block {block.index} (Difficulty: {difficulty})")
+
+    while True:
+
+        # Check max_nonce limit
+        if max_nonce is not None and block.nonce > max_nonce:
+            if logger:
+                logger.warning("Max nonce exceeded during mining.")
+            raise MiningExceededError("Mining failed: max_nonce exceeded")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)